webapp/portlet/src/main/webapp/activity-composer-app/components/ExoActivityComposer.vue

```diff
@@ -61,16 +91,27 @@ export default {
       message: '',
       showErrorMessage: false,
       activityComposerActions: [],
-      attachments: []
+      attachments: [],
+      uploadingCount: 0,
+      percent: 100,
+      actionsData: [],
+      actionsEvents: [],
+      actionComp: [],
     };
   },
   computed: {
     postDisabled: function() {
       const pureText = this.message ? this.message.replace(/<[^>]*>/g, '').replace(/&nbsp;/g, '').trim() : '';
-      return pureText.length === 0 || pureText.length > this.MESSAGE_MAX_LENGTH;
+      return pureText.length === 0 && this.actionsData['file'].value.length === 0 || pureText.length > this.MESSAGE_MAX_LENGTH || this.uploading;
```

Comment:
This links the "file" plugin to the activity composer. It should not. The activity composer should be agnostic of the plugins.
We should find a way to set the `attachments` variable to `this.actionsData['file'].value` when files are selected. Then we can only work with `this.attachments`. WDYT ?